### ISSUE TITLE
Make Error "‘usin’ undeclared (first use in this function)"

### ISSUE THREAD

**Issue by BlueWaterCrystal** [danluke2/mpudp_vpn_thesis#3]
Get this when making mptcp with the mpudp udp.c

used:
ubuntu 14.04
mptcp_v0.90

`net/ipv4/udp.c: In function ‘udp_sendmsg’:
net/ipv4/udp.c:964:53: error: ‘usin’ undeclared (first use in this function)
   if (mpudp_send_function && (*mpudp_send_function)(usin)){
                                                     ^
net/ipv4/udp.c:964:53: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [net/ipv4/udp.o] Error 1
make[1]: *** [net/ipv4] Error 2
make: *** [net] Error 2
`

**Comment by danluke2** [comment]
Not sure why this error is coming up since 'usin' is declared in the same function body.  I have not tried compiling the code in some time, so give me a couple of days to see if I get the same error

**Comment by BlueWaterCrystal** [comment]
Hey, did you manage to compile the code? 
is MPUDP_recv and MPUDP_send required for compiling, I thought it was only for OpenVPN 

also is it possible to use different udp vpns with it?

**Comment by danluke2** [comment]
they aren't needed to compile.  I am working on it now and ran into some problems since I wanted to see if it would work on newer system/branch (which it didn't).

I hope to have an answer for you later today or tomorrow.  Sorry for the delay

**Comment by BlueWaterCrystal** [comment]
sounds good

**Comment by danluke2** [comment]
Updated the udp.c file to fix the compiler issue.  I was able to compile the kernel without any errors, but I did not do extensive testing to verify this version works like the original did with the thesis work.  

It seems the linux udp.c file was updated with new imports and some code changes that caused the compiler issues when replacing with the original file.  This new version took the main code pieces from the original for MPUDP to work and inserted them into the new udp.c version.
